doc: prepare new release 3.5.0 (#4080)
tracim/tracim: changelog.md

## 3.5.0 / 2021-xx-xx

### New Features

> - All users can now set a public user profil

Reviewer: users have a parameterizable public profile

Author: Done.

> - All users can now set a public user profil

Reviewer 2: ```suggestion
- All users can now set a public user profile
```

Author: Done.

### Fixed Issues

- Frontend: [#3578](https://github.com/tracim/tracim/issues/3578),
[#3785](https://github.com/tracim/tracim/issues/3785),
[#3790](https://github.com/tracim/tracim/issues/3790),
[#3989](https://github.com/tracim/tracim/issues/3989),
[#3999](https://github.com/tracim/tracim/issues/3999)
- Other: [#4062](https://github.com/tracim/tracim/issues/4062)

### Known Issues

> - SQLite: IDs of objects can be re-used on database (issue [#4016](https://github.com/tracim/tracim/issues/4016))

Reviewer: the issue is not that id can be reused (this is the reason for the bug). what is the bug symptom?

Author: I not know how to explain that in 2 lines.

Author: I purpose something like this 
~~~
SQLite: if you delete one user in the database and create a new one, SQlite can assign the same user_id like visible for the old one.
In this case, some information in Tracim is not referer to the good user.
~~~

Thats ok for you @lebouquetin @grignards

Reviewer 3: SQLite: After deleting a content/user/space directly in the database its id can be re-used (see details [here](https://sqlite.org/autoinc.html)). This can lead Tracim to display information referring to the wrong object in the notifications and activity feed.

Author: Done

### Breaking Changes

> - Some of renaming and wording changes in frontend (issue [#3901](https://github.com/tracim/tracim/issues/3901))

Reviewer: Some wording in the user interface are changed

By the way, this is a good question - I'm not sure it is a breaking change, actually. It's not that clear what is breaking and what is not

Reviewer: need probably a discussion on what we consider breaking (or technicall breakin / user breaking)

Author: For trained users, its a breaking change because old naming are unavailable.

> - Some of renaming and wording changes in frontend (issue [#3901](https://github.com/tracim/tracim/issues/3901))

Reviewer 4: ```suggestion
- Wording changes have been done in the frontend (issue [#3901](https://github.com/tracim/tracim/issues/3901))
```

Author: I use the proposal "Some wording in the user interface are changed" because its more clear to use user interface in place of frontend.

### Others

> - Add backend code to create connection between users (follow user feature) (issue [#3916](https://github.com/tracim/tracim/issues/3916))

Reviewer: Implemented an API for connecting users

Author: Done.

> - Running tests (backend/frontend) should be easier on the development environment installation (issue [#4020](https://github.com/tracim/tracim/issues/4020))

Reviewer: simplified way to run tests on development environment

Author: Done.

# 3.5.0 / 2021-xx-xx

### New Features

- All users have a parameterizable public profile

> - User interface now use font awesome 5+ icon

Reviewer 4: ```suggestion
- The user interface now uses icons from Font Awesome 5
```

### Fixed Issues

- Frontend: [#3578](https://github.com/tracim/tracim/issues/3578),
[#3785](https://github.com/tracim/tracim/issues/3785),
[#3790](https://github.com/tracim/tracim/issues/3790),
[#3989](https://github.com/tracim/tracim/issues/3989),
[#3999](https://github.com/tracim/tracim/issues/3999),
[#4109](https://github.com/tracim/tracim/issues/4109)
- Backend: [#4137](https://github.com/tracim/tracim/issues/4137),
[#4142](https://github.com/tracim/tracim/issues/4142)
- Other: [#4062](https://github.com/tracim/tracim/issues/4062)

### Known Issues

- SQLite: After deleting a content/user/space directly in the database its id can be re-used (see details [here](https://sqlite.org/autoinc.html)). This can lead Tracim to display information referring to the wrong content/user/space in the notification wall and activity feed. (issue [#4016](https://github.com/tracim/tracim/issues/4016))

### Breaking Changes

> - Some wording in the user interface are changed (issue [#3901](https://github.com/tracim/tracim/issues/3901), [#4113](https://github.com/tracim/tracim/issues/4113), [#4114](https://github.com/tracim/tracim/issues/4114))

Reviewer 4: ```suggestion
- Some wordings in the user interface are changed (issue [#3901](https://github.com/tracim/tracim/issues/3901), [#4113](https://github.com/tracim/tracim/issues/4113), [#4114](https://github.com/tracim/tracim/issues/4114))
```

### Others

> - Implemented an API for connecting users (follow user feature) (issue [#3916](https://github.com/tracim/tracim/issues/3916))

Reviewer 4: ```suggestion
- Implemented an API to allow users to follow each other (issue [#3916](https://github.com/tracim/tracim/issues/3916))
```

> - Simplified way to run tests (backend/frontend) on development environment (issue [#4020](https://github.com/tracim/tracim/issues/4020))

Reviewer 4: ```suggestion
- Simplified way to run tests (backend/frontend) in development environments (issue [#4020](https://github.com/tracim/tracim/issues/4020))
```

- Frontend: [#3578](https://github.com/tracim/tracim/issues/3578),
[#3785](https://github.com/tracim/tracim/issues/3785),
[#3790](https://github.com/tracim/tracim/issues/3790),
[#3989](https://github.com/tracim/tracim/issues/3989),
[#3999](https://github.com/tracim/tracim/issues/3999),
[#4109](https://github.com/tracim/tracim/issues/4109)
- Backend: [#4130](https://github.com/tracim/tracim/issues/4130),
[#4137](https://github.com/tracim/tracim/issues/4137),
[#4142](https://github.com/tracim/tracim/issues/4142)
- Other: [#4062](https://github.com/tracim/tracim/issues/4062)

### Known Issues

- SQLite: After deleting a content/user/space directly in the database its id can be re-used (see details [here](https://sqlite.org/autoinc.html)). This can lead Tracim to display information referring to the wrong content/user/space in the notification wall and activity feed. (issue [#4016](https://github.com/tracim/tracim/issues/4016))

### Breaking Changes

- Some wordings in the user interface are changed (issue [#3901](https://github.com/tracim/tracim/issues/3901), [#4113](https://github.com/tracim/tracim/issues/4113), [#4114](https://github.com/tracim/tracim/issues/4114))

> - On existing Tracim installation running with docker image: It is necessary to update your development.ini (use this file [development.ini.sample](.backend/development.ini.sample) to compare). This three new parameters need to be added with this default path on your development.ini:

Reviewer 4: ```suggestion
- On existing Tracim installations using the docker image: It is necessary to update your development.ini (use this file [development.ini.sample](.backend/development.ini.sample) to compare). These three new parameters need to be added with this default path on your development.ini:
```
